lib/normalizer.c

```diff
@@ -1892,20 +1918,24 @@ grn_nfkc_normalize_unify_katakana_trailing_o(grn_ctx *ctx,
   return current;
 }
 
+typedef grn_bool
+(*grn_nfkc_normalize_is_target_char_func)(const unsigned char *utf8_char,
+                                          size_t length);
+
 static const unsigned char *
-grn_nfkc_normalize_unify_kana_prolonged_sound_mark(grn_ctx *ctx,
-                                                   const unsigned char *start,
-                                                   const unsigned char *current,
-                                                   const unsigned char *end,
-                                                   size_t *n_used_bytes,
-                                                   size_t *n_used_characters,
-                                                   unsigned char *unified_buffer,
-                                                   size_t *n_unified_bytes,
-                                                   size_t *n_unified_characters,
-                                                   void *user_data)
+grn_nfkc_normalize_unify_to_previous_kana_vowel_or_n(grn_ctx *ctx,
+                                                     const unsigned char *start,
+                                                     const unsigned char *current,
+                                                     const unsigned char *end,
+                                                     size_t *n_used_bytes,
+                                                     size_t *n_used_characters,
+                                                     unsigned char *unified_buffer,
+                                                     size_t *n_unified_bytes,
+                                                     size_t *n_unified_characters,
+                                                     grn_nfkc_normalize_is_target_char_func func,
```

Comment:
How about using `user_data` instead of adding a new argument?

```c
typedef struct {
  grn_nfkc_normalize_is_target_char_func is_prolonged_sound_mark;
  bool previous_length;
} grn_nfkc_normalize_prolonged_sound_mark_like_data;

static const unsigned char *
grn_nfkc_normalize_unify_kana_prolonged_sound_mark_like(...)
{
  grn_nfkc_normalize_prolonged_sound_mark_like_data data = user_data;
  ...
  if (data->previous_length == 3 && data->is_prolonged_sound_mark(current, char_length)) {
    ...
  }
}
```

Reply:
Sounds good.
I have added new `struct` used as `user_data`.